scene/gui/color_picker.cpp

```diff
@@ -1400,6 +1465,85 @@ void ColorPicker::_uv_input(const Ref<InputEvent> &p_event, Control *c) {
 			emit_signal(SNAME("color_changed"), color);
 		}
 	}
+
+	// TODO: Is it better to cast and check it or to not check it and get vector?
+	Ref<InputEventKey> kev = p_event;
+	Ref<InputEventJoypadButton> jbev = p_event;
+	Ref<InputEventJoypadMotion> jmev = p_event;
+
+	if (kev.is_valid() || jbev.is_valid() || jmev.is_valid()) {
+		// TODO: Consider adding new ui actions specific to ColorPicker, like the ones used for LineEdit
+		Vector2 color_change_vector = Vector2();
+		if (p_event->is_action_pressed("ui_left", true)) {
+			color_change_vector.x -= 1;
+		} else if (p_event->is_action_pressed("ui_right", true)) {
+			color_change_vector.x += 1;
+		} else if (p_event->is_action_pressed("ui_up", true)) {
+			color_change_vector.y -= 1;
+		} else if (p_event->is_action_pressed("ui_down", true)) {
+			color_change_vector.y += 1;
+		}
+
+		if (!Math::is_zero_approx(color_change_vector.length())) {
```

Comment:
```suggestion
		if (!Math::is_zero_approx(color_change_vector.length_sqared())) {
```
No need for a square root here

Reply:
I realized I could make it even easier and more readable and changed it to `!color_change_vector.is_zero_approx()`